### PR TITLE
Correctly trim position info

### PR DESCRIPTION
**Pull request by siefkenj**
The previous `filterPositionInfo` didn't recuse properly. This new one should :-)

### CODE REVIEW

File: packages/parser/src/dast-to-xml/utils.ts [hunk 1]
```diff
@@ -87,19 +87,17 @@ export function mergeAdjacentTextInArray(nodes: DastNodes[]): DastNodes[] {
 export function filterPositionInfo(
     nodes: DastNodes | DastNodes[],
 ): DastNodes | DastNodes[] {
+    if (typeof nodes !== "object" || nodes === null) {
+        return nodes;
+    }
     if (Array.isArray(nodes)) {
         return nodes.flatMap(filterPositionInfo);
     }
     if (nodes && typeof nodes === "object" && "position" in nodes) {
```
[the remark this answers is not in the document]
Comment (siefkenj) [reply packages/parser/src/dast-to-xml/utils.ts:96]: Where would the early return be triggered? If I remove that if statement, I get an infinite loop.

[the remark this answers is not in the document]
Comment (dqnykamp) [reply packages/parser/src/dast-to-xml/utils.ts:96]: Am I just thinking about this wrong? If `typeof nodes` isn't "object", then it should return on line 91. If nodes==null or nodes==undefined or nodes==[most any other falsy], it should return on line 91.

[the remark this answers is not in the document]
Comment (siefkenj) [reply packages/parser/src/dast-to-xml/utils.ts:96]: Are you referring to the new line `91` or the old line `91`?

[the remark this answers is not in the document]
Comment (dqnykamp) [reply packages/parser/src/dast-to-xml/utils.ts:96]: new line 91

[the remark this answers is not in the document]
Comment (siefkenj) [reply packages/parser/src/dast-to-xml/utils.ts:96]: Okay. I understand your comment now :-)

**Review by dqnykamp — APPROVED** [review]

**Pull request merged**